atsc/src/header.rs

```diff
@@ -14,25 +14,106 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-use bincode::{Decode, Encode};
+use log::{debug, trace};
+use std::panic;
 
-#[derive(Encode, Decode, Debug, Clone)]
+/*  The current file version.
+    On file read, compressors check the version and uncompress accordingly to that (of fail)
+*/
+const CURRENT_VERSION: u32 = 1;
+#[derive(Debug, Clone)]
 pub struct CompressorHeader {
     initial_segment: [u8; 4],
-    // We should go unsigned
-    frame_count: i16,
+    pub version: u32,
+    frame_count: u8,
+}
+
+fn verify_header_versions(version: u32) {
+    let current_version = CURRENT_VERSION;
+    trace!("Versions: c:{} h:{}", current_version, version);
+    match current_version.cmp(&version) {
+        std::cmp::Ordering::Less => panic!(
+            "Can't decompress! File is version ({}) is higher than compressor version ({})!",
+            version, current_version
+        ),
+        std::cmp::Ordering::Equal | std::cmp::Ordering::Greater => {
+            debug!("File version: {}", version)
+        }
+    }
 }
 
 impl CompressorHeader {
     pub fn new() -> Self {
         CompressorHeader {
             initial_segment: *b"BRRO",
-            // We have to limit the bytes of the header
+            version: CURRENT_VERSION,
             frame_count: 0,
         }
     }
 
     pub fn add_frame(&mut self) {
         self.frame_count += 1;
     }
+
+    pub fn get_frame_count(&mut self) -> u8 {
+        self.frame_count
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
```

Comment:
To reduce allocations I would define this as:
`pub fn to_bytes(&self, writer: &mut Vec<u8>)`
And then write to the passed in vec instead of creating a new one.